packages/web-components/src/components/ic-select/ic-select.spec.tsx

```diff
@@ -186,248 +169,642 @@ describe("ic-select", () => {
+  it("should render as required", async () => {
+    const page = await newSpecPage({
+      components: [Select, Menu, InputComponentContainer],
+      html: getTestSelect('required="true" searchable="true"'),
+    });
+
+    expect(page.root).toMatchSnapshot("required-searchable");
+  });
+
+  it("should test with clear button with searchable", async () => {
+    const page = await newSpecPage({
+      components: [Select, Menu, InputComponentContainer, Button],
+      html: `<ic-select label="IC Select Test" searchable="true"></ic-select>`,
+    });
+    page.root.options = menuOptions;
+    page.rootInstance.searchableSelectInputValue = "test value";
+    await page.waitForChanges();
+
+    const clearButton = page.root.shadowRoot.querySelector(
+      clearButtonId
+    ) as HTMLIcButtonElement;
+
+    clearButton.focus();
+    await page.waitForChanges();
+    expect(page.rootInstance.clearButtonFocused).toBe(true);
+
+    clearButton.blur();
+    await page.waitForChanges();
+    expect(page.rootInstance.clearButtonFocused).toBe(false);
+
+    clearButton.click();
+    await page.waitForChanges();
+    expect(page.rootInstance.searchableSelectInputValue).toBeNull;
+  });
+
+  it("should test keydown handler searchable", async () => {
+    const page = await newSpecPage({
+      components: [Select, Menu, InputComponentContainer],
+      html: `<ic-select label="IC Select Test" searchable="true"></ic-select>`,
+    });
+    jest.spyOn(page.rootInstance, "setFocus").mockImplementation();
+    page.root.options = menuOptions;
+    await page.waitForChanges();
+
+    await page.root.setFocus();
+    await page.waitForChanges();
+
+    await page.rootInstance.handleKeyDown({
+      key: "ArrowDown",
+      preventDefault: (): void => null,
+    });
+    await page.waitForChanges();
+
+    const menu = page.root.shadowRoot.querySelector("ic-menu");
+    expect(menu.options).toEqual(menuOptions);
+    expect(page.rootInstance.open).toBeTruthy;
+
+    await page.waitForChanges();
+    await page.rootInstance.handleKeyDown({
+      key: "Enter",
+      preventDefault: (): void => null,
+    });
+    await page.waitForChanges();
+
+    expect(page.rootInstance.open).toBeFalsy;
+
+    page.rootInstance.open = true;
+    page.rootInstance.noOptions = true;
+    await page.waitForChanges();
+    await page.rootInstance.handleKeyDown({
+      key: "Ctrl",
+      preventDefault: (): void => null,
+    });
+    await page.waitForChanges();
+
+    expect(page.rootInstance.setFocus).toHaveBeenCalledTimes(2);
+
+    page.rootInstance.open = true;
+    page.rootInstance.noOptions = true;
+    await page.waitForChanges();
+    await page.rootInstance.handleKeyDown({
+      key: "Backspace",
+      preventDefault: (): void => null,
+    });
+    await page.waitForChanges();
+
+    expect(page.rootInstance.setFocus).toHaveBeenCalledTimes(3);
+  });
+
+  it("should test menu handleKeyboardOpen method - arrow up", async () => {
```

Comment:
Isolated menu tests have been created in the ic-menu.spec, I don't see that menu in conjunction with select does any harm but could possibly be removed if considered redundant?